crates/api_models/src/events/payment.rs

```diff
@@ -62,6 +63,16 @@ impl ApiEventMetric for PaymentsCompleteAuthorizeRequest {
     }
 }
 
+impl ApiEventMetric for PaymentsDynamicTaxCalculationRequest {
+    fn get_api_event_type(&self) -> Option<ApiEventsType> {
+        Some(ApiEventsType::Payment {
+            payment_id: self.payment_id.clone(),
+        })
```

Comment:
Is this part of the payment lifecycle ?,
if not this can be none

Reply:
Yes this is part of payment life cycle.